### PR TITLE
control-service: Integration test for listing many jobs

### PR DESCRIPTION
**Pull request by gageorgiev**
This PR introduces an integration test covering the case
where many (in this case 50) jobs are created and then
listed, checking that they all appear in the listing. This is to
check for a bug where occasionally listing jobs won't print
all of them.
Issue https://github.com/vmware/versatile-data-kit/issues/1238

Testing done: introduced integration test

Signed-off-by: Gabriel Georgiev <gageorgiev@vmware.com>

### CODE REVIEW

**Comment by murphp15** [comment]
so does this fail sometimes?

**Comment by gageorgiev** [comment]
> so does this fail sometimes?

It should have, but it doesn't seem to have failed this time.

**Comment by murphp15** [comment]
IMO we should keep working on this test till it can reproduce the issue quite often. 
At the moment I think it brings little value.

File: projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  private static final int JOB_COUNT = 50;
+
+  @Test
+  public void testListManyJobs() throws Exception {
+    createDummyJobs();
+
+    mockMvc
+        .perform(
+            get(String.format("/data-jobs/for-team/%s/jobs", TEST_TEAM_NAME))
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java:42]: could you list the tests that cover this endpoint already?
also, what is the diff between those and these newly added ones, the heavier JOB_COUNT?

File: projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                        + "\"filter\": ["
+                        + "    {"
+                        + "      \"property\": \"config.team\","
+                        + "      \"pattern\": \""
+                        + TEST_TEAM_NAME
+                        + "\""
+                        + "    }"
+                        + "  ],"
+                        + "\"pageNumber\": 1,"
+                        + "\"pageSize\": "
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java:57]: does the issue has anything to do with page size? what is the default page size currently, 50? is that why you create 50 jobs, to check the paging on page 2? could you then adjust the page size filter query?

File: projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java (hunk 1 above)
Comment (ivakoleva) [reply projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java:42]: could you also check the cicd, if those tests already covering the endpoint had failed while listing jobs? if yes, could you apply the pipeline jobs, so we could analyse the common denominator to find the cause?

**Comment by ivakoleva** [comment]
Is there a ticket related to this PR? if yes, you  should link it.

**Comment by ivakoleva** [comment]
> Is there a ticket related to this PR? if yes, you should link it.

Found it https://github.com/vmware/versatile-data-kit/issues/1238
You can ask the issue reporter to show if reproducible, in case you face a problem to do so by yourself. If neither can reproduce it, you should close the ticket/PR

**Comment by gageorgiev** [comment]
Closing this as the target bug cannot be reproduced both manually and through this test. This PR is linked in the closed issue so if this bug appears again in the future, we can reopen this PR.

**Pull request closed without merging**